[#134] Use the combination of hostname + timestamp for a heap dump

iperdomo opened the pull request:
We can configure the JVM to generate a heap dump when an
OutOfMemoryError happens.  The parameter -XX:HeapDumpPath can be a
folder or a file.

When is a folder the JVM generates a java_pid<pid>.hprof file.
In the case of our services the java process running in a container we
always have the PID 1, which means that the path to the file is the
same.

The JVM will refuse to overwrite a file if this already exists,
e.g. Unable to create /dumps/java_pid1.hprof: File exists

By concatenating the container hostname and timestamp we generate
a unique filename for the heap dump.

iperdomo committed:
[#134] Use the combination of hostname + timestamp for a heap dump

We can configure the JVM to generate a heap dump when an
OutOfMemoryError happens.  The parameter -XX:HeapDumpPath can be a
folder or a file.

When is a folder the JVM generates a java_pid<pid>.hprof file.
In the case of our services the java process running in a container we
always have the PID 1, which means that the path to the file is the
same.

The JVM will refuse to overwrite a file if this already exists,
e.g. Unable to create /dumps/java_pid1.hprof: File exists

By concatenating the container hostname and timestamp we generate
a unique filename for the heap dump.

dlebrero commented:
I have seen systems to go down because a strategy like this, as it means an unbounded amount of dumps that end up filling up the disk. 

Can we delete the dump after it has been copied to google storage?

iperdomo commented:
> Can we delete the dump after it has been copied to google storage?

Sure

iperdomo committed:
[#134] Delete dump file after uploading to Storage bucket

Trying to avoid filling up the disk with an ever growing number of
heap dump files.
